sprokit/processes/core/crop_detections_process.h

```diff
@@ -0,0 +1,79 @@
+/*ckwg +29
+ * Copyright 2017 by Kitware, Inc.
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ *  * Redistributions of source code must retain the above copyright notice,
+ *    this list of conditions and the following disclaimer.
+ *
+ *  * Redistributions in binary form must reproduce the above copyright notice,
+ *    this list of conditions and the following disclaimer in the documentation
+ *    and/or other materials provided with the distribution.
+ *
+ *  * Neither name of Kitware, Inc. nor the names of any contributors may be used
+ *    to endorse or promote products derived from this software without specific
+ *    prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE AUTHORS OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#ifndef KWIVER_CROP_DETECTIONS_PROCESS_H_
+#define KWIVER_CROP_DETECTIONS_PROCESS_H_
+
+#include <sprokit/pipeline/process.h>
+
+#include "kwiver_processes_export.h"
+
+#include <memory>
+
+namespace kwiver
+{
+
+// ----------------------------------------------------------------
+/**
+ * \class crop_detections_process
+ *
+ * \brief Splits an image into multiple chips. One for each detection.
+ *
+ * \iports
+ * \iport{image}
+ *
```

Comment:
Some more documentation for other input ports? (e.g. DetectedObjectSet)

Reply:
I agree with the need for documentation, but I will add that I put this documentation in the .cxx file rather than the header for processes only. The header for a process is so thin and nobody really looks at it. I believe that if this process documentation is in the .cxx file, it may be read and updated more often.

Reply:
OK, but what about installations of KWIVER?  Users from that perspective are not going to have CXX files to read.  Are we not caring about that yet?

Reply:
That's a good point. It could go back in the header. We could ship the documentation which is easier to read than the original markup.